epitator/annospan.py

```diff
@@ -140,3 +134,37 @@ def iterate_leaf_base_spans(self):
         for span in self.iterate_base_spans():
             if not isinstance(span, SpanGroup):
                 yield span
+
+    def combined_metadata(self):
+        """
+        Return the merged metadata dictionaries from all descendant spans.
+        Presedence of matching properties follows the order of a pre-order tree traversal.
+        """
+        leaf_spans = list(self.iterate_base_spans())
+        leaf_spans.reverse()
+        result = {}
+        for leaf_span in leaf_spans + [self]:
+            if leaf_span.metadata:
+                result.update(leaf_span.metadata)
+        return result
```

Comment:
I see that point about the usage of `@property`. I was partially using it because it would allow for different ways of constructing a `span.metadata` property which behaved to legacy code like a metadata dict, but was constructed in more complex ways.